src/modules/ui/organisms/AddConfigurationForm/AddCredentialsForm.tsx

```diff
@@ -0,0 +1,179 @@
+import { Button, Box } from "@saleor/macaw-ui/next";
+import { type SubmitHandler, useFormContext } from "react-hook-form";
+import { useCallback, useEffect } from "react";
+import { useAppBridge } from "@saleor/app-sdk/app-bridge";
+import { useRouter } from "next/router";
+import { RoundedBoxWithFooter } from "../../atoms/RoundedActionBox/RoundedActionBox";
+import { FormInput } from "@/modules/ui/atoms/macaw-ui/FormInput";
+import { trpcClient } from "@/modules/trpc/trpc-client";
+import { getErrorHandler, getFieldErrorHandler, getFormFields } from "@/modules/trpc/utils";
+import { invariant } from "@/lib/invariant";
+import { type PaymentAppFormConfigEntry } from "@/modules/payment-app-configuration/config-entry";
+import { FormSelect } from "@/modules/ui/atoms/macaw-ui/FormSelect";
+
+const actionId = "payment-form";
+
+export const AddCredentialsForm = ({
+  configurationId,
+}: {
+  configurationId?: string | undefined | null;
+}) => {
+  const formMethods = useFormContext<PaymentAppFormConfigEntry>();
+  const { appBridge } = useAppBridge();
+  const router = useRouter();
+
+  const context = trpcClient.useContext();
+
+  const {
+    handleSubmit,
+    reset,
+    setError,
+    control,
+    formState: { defaultValues },
+  } = formMethods;
+
+  const { data: klarnaConfigurationData } =
+    trpcClient.paymentAppConfigurationRouter.paymentConfig.get.useQuery(
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- query is not enabled if configurationId is missing
+      { configurationId: configurationId! },
+      {
+        enabled: !!configurationId,
+        onError: (err) => {
+          getErrorHandler({
```

Comment:
We should use `getFieldErrorHandler` here so that errors are properly displayed by react-hook-form